src/geometry/manifold/partition_of_unity.lean

```diff
@@ -82,33 +92,132 @@ then a choice of `smooth_bump_covering` is available as `smooth_bump_covering.ch
+
+* all functions are infinitely smooth and nonnegative;
+* the family `λ i, support (f i)` is locally finite;
+* for all `x ∈ s` the sum `∑ᶠ i, f i x` equals one;
+* for all `x`, the sum `∑ᶠ i, f i x` is less than or equal to one. -/
+structure smooth_partition_of_unity (s : set M := univ) :=
+(to_fun : ι → C^∞⟮I, M; 𝓘(ℝ), ℝ⟯)
+(locally_finite' : locally_finite (λ i, support (to_fun i)))
+(nonneg' : ∀ i x, 0 ≤ to_fun i x)
+(sum_eq_one' : ∀ x ∈ s, ∑ᶠ i, to_fun i x = 1)
+(sum_le_one' : ∀ x, ∑ᶠ i, to_fun i x ≤ 1)
+
+variables {ι I M}
+
+namespace smooth_partition_of_unity
+
+variables {s : set M} (f : smooth_partition_of_unity ι I M s)
+
+instance {s : set M} : has_coe_to_fun (smooth_partition_of_unity ι I M s) :=
+⟨λ _, ι → C^∞⟮I, M; 𝓘(ℝ), ℝ⟯, smooth_partition_of_unity.to_fun⟩
+
+protected lemma locally_finite : locally_finite (λ i, support (f i)) :=
+f.locally_finite'
+
+lemma nonneg (i : ι) (x : M) : 0 ≤ f i x := f.nonneg' i x
+
+lemma sum_eq_one {x} (hx : x ∈ s) : ∑ᶠ i, f i x = 1 := f.sum_eq_one' x hx
+
+lemma sum_le_one (x : M) : ∑ᶠ i, f i x ≤ 1 := f.sum_le_one' x
+
+/-- Reinterpret a smooth partition of unity as a continuous partition of unity. -/
+def to_partition_of_unity : partition_of_unity ι M s :=
+{ to_fun := λ i, f i, .. f }
+
+lemma smooth_sum : smooth I 𝓘(ℝ) (λ x, ∑ᶠ i, f i x) :=
+smooth_finsum (λ i, (f i).smooth) f.locally_finite
+
+lemma le_one (i : ι) (x : M) : f i x ≤ 1 := f.to_partition_of_unity.le_one i x
+
+lemma sum_nonneg (x : M) : 0 ≤ ∑ᶠ i, f i x := f.to_partition_of_unity.sum_nonneg x
+
+/-- A smooth partition of unity `f i` is subordinate to a family of sets `U i` indexed by the same
+type if for each `i` the closure of the support of `f i` is a subset of `U i`. -/
+def is_subordinate (f : smooth_partition_of_unity ι I M s) (U : ι → set M) :=
+∀ i, closure (support (f i)) ⊆ U i
+
+@[simp] lemma is_subordinate_to_partition_of_unity {f : smooth_partition_of_unity ι I M s}
+  {U : ι → set M} :
+  f.to_partition_of_unity.is_subordinate U ↔ f.is_subordinate U :=
+iff.rfl
+
+alias is_subordinate_to_partition_of_unity ↔
+  _ smooth_partition_of_unity.is_subordinate.to_partition_of_unity
+
+end smooth_partition_of_unity
+
+namespace bump_covering
+
+-- Repeat variables to drop two of them
+lemma smooth_to_partition_of_unity {E : Type uE} [normed_group E] [normed_space ℝ E]
+  {H : Type uH} [topological_space H] {I : model_with_corners ℝ E H}
+  {M : Type uM} [topological_space M] [charted_space H M] {s : set M}
+  (f : bump_covering ι M s) (hf : ∀ i, smooth I 𝓘(ℝ) (f i)) (i : ι) :
+  smooth I 𝓘(ℝ) (f.to_partition_of_unity i) :=
+(hf i).mul $ smooth_finprod_cond (λ j _, smooth_const.sub (hf j)) $
+  by { simp only [mul_support_one_sub], exact f.locally_finite }
+
+variables {s : set M}
+
+/-- A `bump_covering` such that all functions in this covering are smooth generates a smooth
+partition of unity. -/
+def to_smooth_partition_of_unity (f : bump_covering ι M s) (hf : ∀ i, smooth I 𝓘(ℝ) (f i)) :
```

Comment:
this one feels strange. Why not define instead `to_smooth_bump_covering`, and then also a function from smooth bump coverings to smooth partitions of unity? ok, I understand that you use this one below to do precisely this. Can you then explain in the docstring that it is a shortcut for `to_smooth_bump_covering.to_smooth_partition_of_unity`, used to construct the latter?

Reply:
Added an explanation.